sdccc/src/main/java/com/draeger/medical/sdccc/manipulation/precondition/impl/ManipulationPreconditions.java

```diff
@@ -431,6 +450,50 @@ static boolean verifyStatePresentAndAssociated(
 
             return valid;
         }
+
+        static final class EpisodicContextReportStateHandleObserver implements MdibAccessObserver {
+
+            private final String expectedHandle;
+            private final Object signal = new Object();
+
+            private EpisodicContextReportStateHandleObserver(final String expectedStateHandle) {
+                this.expectedHandle = expectedStateHandle;
+            }
+
+            @Subscribe
+            public void onUpdate(final ContextStateModificationMessage report) {
+                for (List<AbstractContextState> states : report.getStates().values()) {
+                    for (AbstractContextState state : states) {
+                        if (this.expectedHandle.equals(state.getHandle())) {
+                            synchronized (signal) {
+                                this.signal.notifyAll();
+                            }
+                        }
+                    }
+                }
+            }
+
+            public void waitForStateUpdate(final long timeoutMillis) {
+                long now = System.currentTimeMillis();
+                final long end = now + timeoutMillis;
+                boolean done = false;
+                synchronized (signal) {
+                    while (now < end && !done) {
```

Comment:
this does not guard against spurious wakeup, because "done" is set to "true" for spurious wakeups